File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.py

def solution(number, k):
    answer = []
    
    for num in number:
        if not answer:
            answer.append(num)
            continue
        if k>0:
            while answer[-1] < num:
                answer.pop()
                k -= 1 
                if not answer or k<=0:
                    break
        answer.append(num)
        
    return ''.join(answer[:len(answer)-k])    

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string number, int k) {
    string answer = "";
    answer.push_back(number[0]);
    
    for(int i = 1; i < number.length(); i++) {
        char num = number[i];
        
        while(!answer.empty() && k > 0 && answer.back() < num) {
            answer.pop_back();
            k--;
        }
        answer.push_back(num);
    }
    
    while(k > 0)
    {
        answer.pop_back();
        k--; 
    }
    
    return answer;
}